R/epidist_utils.R

```diff
@@ -377,50 +383,45 @@ create_epidist_summary_stats <- function(mean = NA_real_,
 #' )
 create_epidist_citation <- function(author = NA_character_,
                                     year = NA_integer_,
-                                    PMID = NA_character_,
-                                    DOI = NA_character_) {
+                                    title = NA_character_,
+                                    journal = NA_character_,
+                                    DOI = NA_character_,
+                                    PMID = NA_character_) {
   # check input
   checkmate::assert_character(author)
   checkmate::assert_number(year, na.ok = TRUE)
+  checkmate::assert_character(title)
+  checkmate::assert_character(journal)
   checkmate::assert_number(PMID, na.ok = TRUE)
   checkmate::assert_character(DOI)
 
-  if (is.na(author) || is.na(year) || is.na(DOI)) {
+  if (is.na(author) || is.na(year) || is.na(journal) || is.na(title)) {
     message(
-      "Citation cannot be created as either author, year or DOI is missing"
+      "Citation cannot be created as author, year, journal or title is missing"
     )
```

Comment:
The examples here and on L221-L225 of `epidist.R` needs to be updated accordingly:

https://github.com/epiverse-trace/epiparameter/blob/bfc8ee7cbd107ef977e53dcef0bb1b282aa71cf2/R/epidist_utils.R#L379-L383

https://github.com/epiverse-trace/epiparameter/blob/bfc8ee7cbd107ef977e53dcef0bb1b282aa71cf2/R/epidist.R#L221-L225